Загружаю файлы

stasokulov opened the pull request.

stasokulov committed:
Загружаю файлы

stasokulov committed:
Привет классы в соответствие с БЭМ

6thSence commented on source/script/git.js:
```diff
@@ -0,0 +1,148 @@
+//Подтягиваю свои репозитории с GitHub
```
Комментарии такого рода избыточны в проекте
Что происходит должно быть понятно по самому коду и названиям переменных

6thSence commented on source/script/git.js:
```diff
@@ -0,0 +1,148 @@
+const username = 'stasokulov';
+let linkToMyGit = `https://api.github.com/users/${username}`;
+let numRepos = '';
+let perPage = 10; //Репозиториев на странице
```
лишний комментарий 
лучше дать переменной более говорящее название reposPerPage

6thSence commented on source/script/git.js:
```diff
@@ -0,0 +1,148 @@
+getRepos(page);
+goPagination(); //Включаю слушатель кликов по меню пагинации
+
+//Запрашиваю данные репозиториев
```
нужно убрать все комментарии

6thSence commented on source/script/git.js:
```diff
@@ -0,0 +1,148 @@
+    .then(arr => {
+        //Обрабатываю каждый элемент массива
+        arr.forEach( (element) => {
+            gitItemCreator(element); //Создаю элемент в списке репозиториев
```
название методов должно быть глаголами
если что-то создается, то скорей всего название должно начинаться со слова create

6thSence commented on source/script/git.js:
```diff
@@ -0,0 +1,148 @@
+    })
+    .then( () => {
+        //Убираем прелоадер
+        preloader.classList.add('invis');
```
Чаще для скрытия элементов встречается слово hidden

6thSence commented on source/script/header.js:
```diff
@@ -0,0 +1,44 @@
+    burgerItem2.classList.toggle('burger__item-2_active');
+    mobileMenu.classList.toggle('invis');
+    //language.classList.toggle('header__mainMenu__language_active');
+    //language.classList.toggle('invisLanguage');
```
комментарий кода не должно быть

6thSence commented on source/style/header.css:
```diff
@@ -0,0 +1,91 @@
+        color: var(--gray);
+        display: flex;
+        justify-content: space-between;
+        /*width: 474px;*/
```
убираем

6thSence commented on source/style/main.css:
```diff
@@ -0,0 +1,249 @@
+}
+
+h1, h2 {
+    margin: 0;
```
Обнуление всех свойств не есть хорошая практика
Потому что так нарушается изолированность свойств
и если будут добавляться элементы из внешних модулей или библиотек, у которых использовались стандартные значения, то они не будут ожидать что их меняют и все может повести себя не по плану

поэтому не стоит использовать какие-то глобальные сбросы вообще 

особенно селекторы по теегам, используем только селекторы по классам

6thSence commented on source/style/main.css:
```diff
@@ -0,0 +1,249 @@
+    margin: 0;
+}
+
+li {
```
убрать все селекторы по тегам

6thSence commented on source/style/main.css:
```diff
@@ -0,0 +1,249 @@
+}
+
+/*Contact me*/
+.contactMe_mobileMenu {
```
/*Contact me*/
.contactMe_mobileMenu {

Эти две строки одинаковые
задавай себе вопрос - а этот комментарий точно здесь нужен?

6thSence commented on source/style/recentWorks.css:
```diff
@@ -0,0 +1,149 @@
+    &__item-footer {
+        font-weight: normal;
+        font-size: 20px;
+        color: #FFFFFF;
```
просто #fff

6thSence commented on source/templates/footer.mustache:
```diff
@@ -0,0 +1,50 @@
+            </p>
+            <p class="footer__text">
+               Я junior frontend-developer.
+                <br>
```
В разметке не используем <br> и <hr>
https://lk.webheroschool.ru/blog/474243

6thSence replied:
br и hr

6thSence commented on source/templates/git.mustache:
```diff
@@ -0,0 +1,42 @@
+
+    <section class="achievement__main">
+        <ul class="achievement__list">
+            <!--Это невидимый пункт-шаблон-->
```
лишний коммент

6thSence commented on source/templates/git.mustache:
```diff
@@ -0,0 +1,42 @@
+    <section class="achievement__main">
+        <ul class="achievement__list">
+            <!--Это невидимый пункт-шаблон-->
+            <li class="invis git__template">
```
лучше использовать hidden вместо invis

6thSence commented on source/templates/git.mustache:
```diff
@@ -0,0 +1,42 @@
+                <a class="git__item-innerWrap" href="" target="blank"> 
+                    <p class="git__item-title">svg-sprite-localStorage</p>
+                    <p class="git__item-text">
+                        <span class="marker">
```
Если этот элемент пустой, то возможно стоит использовать псевдо-элементы?

6thSence commented on source/templates/header.mustache:
```diff
@@ -0,0 +1,53 @@
+                <a class="menu__works menu__item_mobileMenu scroll" data-target = 'recentWorks'>
+                    Работы
+                </a>
+                <a class="menu__skills menu__item_mobileMenu scroll" data-target = 'skills'>
```
У тебя используются " и ' кавычки. Нужно выбрать какие-то одни и использовать только их

6thSence commented on source/templates/social.mustache:
```diff
@@ -0,0 +1,8 @@
+<menu class="social">
+    <a href = "https://github.com/stasokulov" target="blank">
+        <li class="social__git social__item"></li>
```
li по семантике должны находиться в ul
поэтому либо нужно обернуть в ul
либо не использовать li

stasokulov committed:
Внес правки.

Заменил способ заполнения секции с гит-репозиториями. Было: создавалась копия элемента-шаблона из разметки, наполнялась данными, вставлялась в секцию. Стало: элемент полностью генерируется в JS.
Косметические правки: выравнивание текста и иконок в строке с данными гит-репозитория.
Удалил комментарии из кода.
В HTML сделал все кавычки одного вида - одинарные.

stasokulov committed:
Внес правки

- Оптимизировал svg-файлы
- Добавил margin для правки по замечанию: "Элемент выходит за границы своего родите https://take.ms/l7oBk"
- Поправил margin по замечанию: "Не используем margin-left, top. В этом случае лучше использовать padding https://take.ms/DxeXR"
- Теперь при открытии меню из бургера остальной контент не отображается.
- Добавил минимальную высоту для правки по замечанию: "При обновлении контента (загрузке) страница не должна прыгать. Для этого стоит задать минимальную-высоту или анимировать ее при изменении https://www.loom.com/share/dec800b45c9c4a0a80c393eb56c315ef"

stasokulov committed:
Внес правки

stasokulov committed:
Внес правки
